### PR TITLE
Results of joint debugging 2021 07 29

### PR DESCRIPTION
**Pull request by cvjjm**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,13 +1,14 @@
 from setuptools import setup
 
-pennylane_devices_list=[
+pennylane_devices_list = [
     'synqs.sqs = pennylane_ls:SingleQuditDevice',
-    'synqs.mqs = pennylane_ls:MultiQuditDevice'
+    'synqs.mqs = pennylane_ls:MultiQuditDevice',
+    'synqs.ls = pennylane_ls:FermionDevice',
```
Comment (fretchen) [inline setup.py:6]: should be synqs fs for fermion simulator no ?

File: pennylane_ls/FermionDevice.py [hunk 2]
```diff
@@ -64,67 +75,94 @@ def apply(self, operation, wires, par):
...
+        #np.sort(np.unique(shots, axis=0, return_counts=True, dtype=[('pattern', ), ('probability', )]), order='pattern')
+
+        patterns, probabilities = np.unique(shots, axis=0, return_counts=True)
+        sort_labels = np.argsort(patterns, axis=0)
+        patterns = patterns[sort_labels]
+        probabilities = probabilities[sort_labels]
+
+        return OrderedDict(zip(map(tuple, patterns), probabilities))
+
+    def pre_measure(self):
```
Comment (fretchen) [inline pennylane_ls/FermionDevice.py:135]: no idea what is going on here. I assume that @TheNewYorkerFlute has some strong opinions.

**Comment by fretchen** [comment]
Very cool. A number of questions that I am curious about when I am back.

File: setup.py (hunk 1 above)
Comment (cvjjm) [reply setup.py:6]: Agree. synqs.fs would be nice.

File: pennylane_ls/FermionDevice.py (hunk 2 above)
Comment (cvjjm) [reply pennylane_ls/FermionDevice.py:135]: This is just a conversion to the right data format the PennyLane wants. With this the pennylane.probs() return type is supported by the device and yields correct probabilities.

Comment (RohitPrasadBhatt) [reply pennylane_ls/FermionDevice.py:135]: @cvjjm, I think we can get probs nicely by converting the unique patterns into decimal values and then sorting the decimal values array. I tested it and this nicely gives an ordered dict. The code is:

patterns, probabilities = np.unique(arr, axis=0, return_counts=True)

patterns_decimal_repr=np.packbits(patterns.astype('int32'), axis=1)
patterns_decimal_repr=patterns_decimal_repr.ravel()

sort_labels = np.argsort(patterns_decimal_repr, axis=0)

patterns = patterns[sort_labels]
probabilities = probabilities[sort_labels]

odict=OrderedDict(zip(map(tuple, patterns), probabilities))

Comment (RohitPrasadBhatt) [reply pennylane_ls/FermionDevice.py:135]: And also:
probabilities = probabilities/probabilities.sum()

File: pennylane_ls/FermionDevice.py [hunk 3]
```diff
@@ -21,31 +23,40 @@
...
     }
 
     name = "Fermion Quantum Simulator Simulator plugin"
     pennylane_requires = ">=0.16.0"
-    version = '0.0.1'
+    version = '0.2.0'
     author = "Vladimir and Donald"
 
-    short_name = "synqs.fqe"
+    short_name = "synqs.ls"
```
Comment (TheNewYorkerFlute) [inline pennylane_ls/FermionDevice.py:41]: synqs.fs

File: pennylane_ls/FermionDevice.py (hunk 2 above)
Comment (cvjjm) [reply pennylane_ls/FermionDevice.py:141]: This is a nice solution!

**Comment by cvjjm** [comment]
Feel free to implement the changes you suggested above in this branch before merging.

**Pull request merged**